pkg/webhooks/serviceaccount/serviceaccount.go

```diff
@@ -0,0 +1,222 @@
+package serviceaccount
+
+import (
+	"fmt"
+	admissionv1 "k8s.io/api/admission/v1"
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"net/http"
+	"os"
+	"regexp"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	admissionctl "sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/openshift/managed-cluster-validating-webhooks/pkg/webhooks/utils"
+)
+
+const (
+	WebhookName       string = "serviceaccount-validation"
+	managedNamespaces string = `(^openshift-.*|kube-system)`
+	docString         string = `Managed OpenShift Customers may not delete the service accounts under the managed namespaces: %s`
+)
+
+var (
+	timeout int32 = 2
+	log           = logf.Log.WithName(WebhookName)
+	scope         = admissionregv1.NamespacedScope
+	rules         = []admissionregv1.RuleWithOperations{
+		{
+			Operations: []admissionregv1.OperationType{"DELETE"},
+			Rule: admissionregv1.Rule{
+				APIGroups:   []string{""},
+				APIVersions: []string{"v1"},
+				Resources:   []string{"serviceaccounts"},
+				Scope:       &scope,
+			},
+		},
+	}
+	allowedUsers  []string
+	allowedGroups = []string{
+		"system:serviceaccounts:openshift-backplane-srep",
+	}
+	allowedServiceAccounts = []string{
+		"builder",
+		"default",
+		"deployer",
+	}
+	protectedNamespaces = regexp.MustCompile(managedNamespaces)
+	exceptionNamespaces = []string{
```

Comment:
Could we put those namespaces somewhere else, so it can be used by other webhooks?

Reply:
Do you mean we can abstract the variable in a different file, so all the webhooks can call them?
What I am thinking is if we control the exception list within the webhook, it would be easy for us to tweak this particular webhook.
It would be easy for testing as well IMO.